apps/flo-cli/src/adapters/git/git.repo.ts

```diff
@@ -57,174 +41,199 @@ export const getGitStatus = (directory?: string) =>
 //     detached
 //     prunable gitdir file points to non-existent location
 
-const handleGitErrors = <TArgs extends unknown[], TReturn, TFallback>(
-    options: { fallbackValue: TFallback },
-    callback: (...args: TArgs) => TReturn,
-) => {
-    return (...args: TArgs) => {
-        try {
-            // @TODO: @floydnant lets see, not `await`ing here might be a problem
-            return callback(...args)
-        } catch (error) {
-            const hasMessage = z.object({ message: z.string() }).safeParse(error).success
-            if (hasMessage && (error as { message: string }).message.includes('not a git repository'))
-                return options.fallbackValue
-
-            // console.log(error)
-            // process.exit(1)
-            throw error
+export class GitRepository {
+    /** Do not use this constructor directly, use `.init()` instead */
+    constructor(private sysCallService: SysCallService) {}
+
+    getWorktrees = cacheable((directory: string): Worktree[] => {
+        return transformGitErrors(
+            () => {
+                const output = this.sysCallService.execPipe('git worktree list --porcelain', {
+                    cwd: directory,
+                    stdio: 'pipe',
+                })
+                const worktreeTextBlocks = output.split('\n\n').filter(Boolean)
+                const repoRootDir = this.getRepoRootDir(directory)
+                // @TODO: pass cwd as an option / get from context
```

Comment:
This is different. Here we actually only want the caller cwd, to check if a path is the current one cli was called from.

```suggestion
```